(architecture): add Fabric Management page

pau-hedgehog opened the pull request.

edipascale commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,122 @@
+
+### **K9s**
+- A Kubernetes UI that provides visibility and control over Kubernetes resources, allowing you to manage pods, nodes, services, logs, and Fabric resources efficiently. It includes some helper plugins:
+  - **SSH** – SSH into a fabric switch
```
these are hhfab utils, not k9s - shouldn't they go under fabricator?

edipascale commented:
this looks good @pau-hedgehog, my only concern is that I feel there's a big overlap between the first section in this PR and what you previously added in the architecture overview section, and I wonder whether we should just collapse the two sections into one. That being said, having more documentation is never a bad thing, so I'm not against adding this if you / other people feel that they're better kept separate.

mrbojangles3 commented:
I support adding it. I can see that a user might say, what happens when I do X, Y or Z. They would have those questions answered in the Fabric Management Workflow.

pau-hedgehog commented:
I've made some amendments to try to make both @edipascale and @mrbojangles3 happy ;)

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+### **kubectl & kubectl fabric**
+- [`kubectl`](https://kubernetes.io/docs/reference/kubectl/) is the standard CLI tool for [Kubernetes](https://kubernetes.io).
+- `kubectl fabric` is a plugin that extends `kubectl` with fabric-specific commands and interacts with the Kubernetes API Server.
```
As you're going into that details in k9s section it would be good to describe fabric plugin capability to do inspects, ssh/reboot/reinstall here as well to avoid creating an impression that it's only available through the k9s

pau-hedgehog replied:
I'm adding:

It supports the following commands:
    - **vpc** – Manage VPCs: create, attach, peer, or wipe all related resources
    - **switch** – Manage switches: get IP, SSH, serial, reboot, reinstall
    - **connection** – View connections: management, fabric, vpc-loopback
    - **switchgroup** – Create SwitchGroups
    - **external** – Manage external BGP peers and VPC peering
    - **wiring** – Export fabric wiring diagrams
    - **inspect** – Inspect and troubleshoot fabric resources:
        - `fabric` – Overview of control nodes and switches
        - `switch`, `port` – Status, counters, and usage
        - `server`, `connection` – VPC/external attachments and link details
        - `vpc` – Subnet attachments and reachability
        - `bgp`, `lldp` – Neighbor status checks
        - `ip`, `mac` – Lookup usage across the fabric
        - `access` – Check reachability between VPCs, servers, and IPs

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+---
+
+## **GitOps Functionality (ArgoCD)**
```
It's a bit misleading this way. Please explain GitOps in general and that it could be used with any tools that supports K8s (e.g. ArgoCD, Flux, etc.)

pau-hedgehog replied:
Good point. Changing to:

## **GitOps Integration**

GitOps is a continuous deployment approach where the desired state of a Kubernetes cluster is stored in Git. A GitOps controller monitors this repository and ensures the actual cluster state matches the declared state, using the Kubernetes API.

Hedgehog Fabric is fully compatible with GitOps workflows and can be managed declaratively using any GitOps-compatible tool such as [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) or [Flux](https://fluxcd.io).

- **User Responsibilities:**
    - Define Fabric Custom Resources (CRs) in YAML.
    - Commit and push these CRs to a [Git repository](https://git-scm.com) for version control and collaboration.

- **GitOps Tool Responsibilities:**
    - Monitor the Git repository for updates.
    - Apply changes to the cluster automatically via the Kubernetes API Server.

This enables seamless integration of Hedgehog Fabric into modern CI/CD pipelines and Git-centric workflows.

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+GitOps workflows can be leveraged using [ArgoCD](https://argo-cd.readthedocs.io/en/stable/). This is an alternative approach to show that a Fabric can be used with industry standard tools seamlessly.
+
+- **User Actions:**
```
Could go as an example workflow with ArgoCD

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+The installation of a Hedgehog Fabric deployment is carried out using Fabricator (hhfab CLI). Once deployed, ongoing operations are managed via the [Kubernetes](https://kubernetes.io) CLI, [`kubectl`](https://kubernetes.io/docs/reference/kubectl/).
+
+In this workflow, the Kubernetes API Server processes the Fabric Custom Resources (CRs) and forwards them to the Fabric Controller. The Fabric Controller then translates these high-level network intents into concrete network configurations and manages their application to the underlying infrastructure. The Controller continuously monitors the state of the network resources, ensuring they match th ...
```
API Server isn't forwarding CRs, controllers watches API for CR changes

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+Fabricator -->|Applies CRs| K8S
+KubectlFabric -->|Interacts with| K8S
+K9s -->|UI Manages| K8S
+K8S -->|Sends CRDs| FC
```
API Server isn't forwarding CRs, controllers watches API for CR changes

pau-hedgehog replied:
addressed and updated diagram

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+User -.->|CLI| Fabricator
+User -.->|CLI| Kubectl
+Kubectl -.->|Plugin| KubectlFabric
+Fabricator -->|SSH access| K9s
```
Not sure what is represented by this, could you please elaborate?

pau-hedgehog replied:
This is how you usually access k9s, unless there's another way I'm unaware of:
```
hhfab vlab ssh -n control-1
k9s
```

pau-hedgehog replied:
Removed

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+### **User**
+- **Creates Fabric CR YAMLs** and applies them through standard Kubernetes resource management.
+- **Directly interacts with SONiC switches** via the Fabricator CLI.
```
I guess it's about `hhfab vlab ssh` helper so it should be removed from here. Additionally, we don't allow users to do the changes on the switches

pau-hedgehog replied:
Removed

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+- `kubectl fabric` is a plugin that extends `kubectl` with fabric-specific commands and interacts with the Kubernetes API Server.
+
+### **Fabricator**
+- CLI tool that provides direct interaction with the Kubernetes API.
```
It doesn't provide it, what do you mean?

pau-hedgehog replied:
I was thinking specifically of `hhfab init -w`:
`--wiring FILE, -w FILE [ --wiring FILE, -w FILE ]  include wiring diagram FILE with ext .yaml (any Fabric API objects)`

Frostman commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,102 @@
+
+### **Fabricator**
+- CLI tool that provides direct interaction with the Kubernetes API.
+- Can apply configurations via **SSH access** (using **K9s**) or by directly managing Fabric CRs with YAML files.
```
Kinda same - I guess you're talking about `hhfab vlab ssh` which shouldn't be mentioned outside of the vlab section of the docs

qmonnet commented on docs/architecture/management.md:
```diff
@@ -0,0 +1,127 @@
+
+---
+
+## **Management Workflow Overview**
```
I didn't go through the whole page, this is just a note in passing - I'd strongly recommend _against_ manually formatting all section/subsection headings in bold. The formatting from the theme should be enough already, no need to add manual formatting; if we decided it's not enough, this is something we should change by tuning the CSS for the whole website, for consistency.

qmonnet commented:
I note this PR has been open for a while, what's the status? Is it still pending reviews from the three of us? :see_no_evil:

pau-hedgehog commented:
> I note this PR has been open for a while, what's the status? Is it still pending reviews from the three of us? 🙈

I addressed all comments a few weeks ago. But I understand we have more pressing priorities. Thus I didn't chase anyone

qmonnet reviewed (changes requested):
Nice document!

My high-level comment would be that it's not super-clear how it relates to the “Architecture” section, as it's somewhat of a mix of an architectural-ish description of the management processes and a command reference, with an intro and some headings that might make one believe it will be some user guide to manage the Fabric.

Just to be clear, I do agree your page is in the right section :slightly_smiling_face:, but I'd try to rework it a little bit to orient it even more towards the architecture-description aspect, not sure how, maybe by making clear you introduce the different “actors” from your first Mermaid diagram, and explaining their different interactions - and keep the “reference” parts to a minimum. Maybe run your doc through some LLM, to see if it's got some useful input?

Frostman reviewed (changes requested):
^^
